pkg/cmd/cli/config/store.go

```diff
@@ -116,13 +116,15 @@ func (c *ConfigStore) SaveToFile(credentialsName string, namespace string, clien
 
 	credentials := clientcmdapi.NewAuthInfo()
 	credentials.Token = clientCfg.BearerToken
-	credentials.ClientCertificate = clientCfg.TLSClientConfig.CertFile
-	if len(credentials.ClientCertificate) == 0 {
-		credentials.ClientCertificateData = clientCfg.TLSClientConfig.CertData
-	}
-	credentials.ClientKey = clientCfg.TLSClientConfig.KeyFile
-	if len(credentials.ClientKey) == 0 {
-		credentials.ClientKeyData = clientCfg.TLSClientConfig.KeyData
+	if len(credentials.Token) == 0 {
```

Comment:
@liggitt Alright, thanks.

In my specific scenario here I still have an issue though:

1. Using `osc` as admin (with the `openshift.local.certificates/admin` client cert and key)
2. I want to switch to another user (regular e2e-user)
3. So I do `osc login -u e2e-user` (will exchange username and pass and get a token)
4. But my `clientCfg` was still carrying the client cert and key data (and now the BearerToken that was just exchanged) and will save all three of them to the config file

It actually looks more like a job of the Client instead of the config file store mechanism. E.g., the client should clear cert and key file/data if at some point it's used to exchange a token and store it in `clientCfg.BearerToken` (unless the server actually requires cert *and* token/basic).